src/blob/core.rs

```diff
@@ -30,6 +30,31 @@ where
     validate_data_during_index_regen: bool,
 }
 
+pub(crate) struct WriteResult {
+    dirty_bytes: u64
+}
+
+impl WriteResult {
```

Comment:
Since it's only an internal structure, I think it's ok to declare its fields with `pub` modifier. This will remove the `impl` altogether and reduce the size of the code a bit.

The same is applicable to `DeleteResult`